### PR TITLE
Convert KiwixService.java to kotlin

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
Fixes: #3009 
Parent Issue: #1826 

Converted KiwixService.java to kotlin

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt [hunk 1]
```diff
@@ -248,8 +248,8 @@ class ZimManageViewModel @Inject constructor(
       .subscribe(
         {
           kiwixService.library
-            .retry(5)
-            .subscribe(library::onNext) {
+            ?.retry(5)
+            ?.subscribe(library::onNext) {
```
Comment (gouri-panda) [inline app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt:252]: If it's nullable then we should handle exceptions.

Comment (MohitMaliFtechiz) [reply app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt]: we are fetching from List so nullable condition will not arrived.

File: app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt [hunk 2]
```diff
@@ -150,7 +150,7 @@ class ZimManageViewModel @Inject constructor(
       updateLibraryItems(booksFromDao, downloads, networkLibrary, languages),
       updateLanguagesInDao(networkLibrary, languages),
       updateNetworkStates(),
-      requestsAndConnectivtyChangesToLibraryRequests(networkLibrary),
+      requestsAndConnectivityChangesToLibraryRequests(networkLibrary),
```
Comment (gouri-panda) [inline app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt:153]: Please revert this. This is not related to the ticket.

File: app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt [hunk 3]
```diff
@@ -219,7 +219,7 @@ class ZimManageViewModel @Inject constructor(
     return None
   }
 
-  private fun requestsAndConnectivtyChangesToLibraryRequests(
```
Comment (gouri-panda) [inline app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt:222]: this too.

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/KiwixService.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  @get:GET(LIBRARY_NETWORK_PATH) val library: Single<LibraryNetworkEntity?>
+  @GET fun getMetaLinks(@Url url: String): Observable<MetaLinkNetworkEntity?>
+
+  /******** Helper class that sets up new services  */
+  object ServiceCreator {
+    fun newHacklistService(okHttpClient: OkHttpClient, baseUrl: String): KiwixService {
+      val retrofit = Retrofit.Builder()
+        .baseUrl(baseUrl)
+        .client(okHttpClient)
+        .addConverterFactory(SimpleXmlConverterFactory.create())
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/KiwixService.kt:42]: We should use an alternate converter factory Here.

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/KiwixService.kt [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import retrofit2.http.GET
+import retrofit2.http.Url
+
+interface KiwixService {
+  @get:GET(LIBRARY_NETWORK_PATH) val library: Single<LibraryNetworkEntity?>
+  @GET fun getMetaLinks(@Url url: String): Observable<MetaLinkNetworkEntity?>
+
+  /******** Helper class that sets up new services  */
+  object ServiceCreator {
+    fun newHacklistService(okHttpClient: OkHttpClient, baseUrl: String): KiwixService {
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/KiwixService.kt:38]: [typo] Use `newHackListService`

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/KiwixService.kt (hunk 4 above)
Comment (gouri-panda) [reply core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/KiwixService.kt:42]: This will fix in #3114

File: app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt (hunk 2 above)
Comment (MohitMaliFtechiz) [reply app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt]: hi @gouri-panda ,

done

File: core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/KiwixService.kt (hunk 5 above)
Comment (MohitMaliFtechiz) [reply core/src/main/java/org/kiwix/kiwixmobile/core/data/remote/KiwixService.kt]: done

**Review by gouri-panda — APPROVED** [review]
